src/Microsoft.AspNetCore.WebHooks.Receivers/WebHookOptions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Xml.Linq;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.AspNetCore.WebHooks
+{
+    /// <summary>
+    /// Provides programmatic configuration for the WebHook infrastructure.
+    /// </summary>
+    public class WebHookOptions
+    {
+        /// <summary>
+        /// Gets the list of action parameter <see cref="Type"/>s the
+        /// <see cref="ModelBinding.WebHookHttpContextModelBinder"/> can be associated with. Parameters assignable to
+        /// the given types may be model bound from <see cref="HttpContext.Items"/> entries.
+        /// </summary>
+        /// <value>
+        /// Default collection includes <see cref="IFormCollection"/>, <see cref="JArray"/>, <see cref="JObject"/>, and
+        /// <see cref="XElement"/>.
+        /// </value>
+        public IList<Type> HttpContextItemsTypes { get; } = new List<Type>
```

Comment:
>HttpContextItemsTypes [](start = 27, length = 21)

I don't super know what this is about... I feel like there's a feature here that needs a better explanation

Reply:
This is about resource filters that call input formatters. When they're done they place the deserialized object into type-named locations in `HttpContext.Items` for use in the `WebHookHttpContextModelBinder`. The mechanism overall exists to reduce the number of times the request body is parsed. This exists to avoid leaking random `HttpContext.Items` entries into the actions and to ensure `WebHookModelBindingProvider` configures the `WebHookHttpContextModelBinder` correctly.

What of that information is needed in the doc comments? Or, are you suggesting we do away with either this knob or the overall "read once" mechanism?